components/modal/modal.js

```diff
@@ -9,6 +9,7 @@ import { getClassNamesWithMods } from '../_helpers';
  */
 class Modal extends Component {
   state = {
+    isActive: false,
```

Comment:
Have you considered adding a constructor here to set the state based on props initially:

```js
constructor(props) {
  super(props);

  this.state = {
    isActive: props.active || false,
  };
}
```

?